pkg/render/node.go

```diff
@@ -1484,11 +1893,20 @@ func GetIPv6Pool(pools []operatorv1.IPPool) *operatorv1.IPPool {
 	return nil
 }
 
+// vppDataplaneEnabled returns true if the given Installation enables VPP
+func vppDataplaneEnabled(cr *operatorv1.InstallationSpec) bool {
+	return cr.CalicoNetwork != nil &&
+		cr.CalicoNetwork.LinuxDataplane != nil &&
+		*cr.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneVPP &&
```

Comment:
It's a bit of a red flag to me that we're relying on two API configs to determine if vpp is enabled. the LinuxDataplane option should be the deciding factor as to whether or not the VPP dataplane is being used.

The cr.CalicoNetwork.VPP should probably be defaulted based on the LinuxDataplane option, and validation should be used to make sure that the two are consistent with each other before ever reaching this function.

Reply:
In other words - cr.CalicoNetwork.VPP should never be nil at this point in the code if LinuxDataplane == VPP